flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/typeutils/AvroSchemaConverter.java

```diff
@@ -359,13 +359,13 @@ public static Schema convertToSchema(LogicalType logicalType, int rowTypeCounter
 				// we have to make sure the record name is different in a Schema
 				SchemaBuilder.FieldAssembler<Schema> builder = SchemaBuilder
 					.builder()
-					.record("row_" + rowTypeCounter)
+					.record(rowName)
 					.fields();
-				rowTypeCounter++;
 				for (int i = 0; i < rowType.getFieldCount(); i++) {
+					String fieldName = fieldNames.get(i);
 					builder = builder
-						.name(fieldNames.get(i))
-						.type(convertToSchema(rowType.getTypeAt(i), rowTypeCounter))
+						.name(fieldName)
+						.type(convertToSchema(rowType.getTypeAt(i), fieldName))
```

Comment:
I think this doesn't work. Because field names may conflict. We may need to concat the parent row name and the field name as the child name.

Reply:
It works fine after test when the same field name in each different row type. Because fields in different namespaces.

Reply:
However, if the parant field name is the same to the child name, there still will be an avro exception. For example, the following example is still failed:

```java
	@Test
	public void test() {
		RowType rowType = (RowType) TableSchema.builder()
			.field("row1", DataTypes.ROW(DataTypes.FIELD("a", DataTypes.STRING())))
			.field("row2", DataTypes.ROW(DataTypes.FIELD("b", DataTypes.STRING())))
			.field("row3", DataTypes.ROW(
				DataTypes.FIELD("row3", DataTypes.ROW(DataTypes.FIELD("c", DataTypes.STRING())))))
			.build().toRowDataType().getLogicalType();
		Schema schema = AvroSchemaConverter.convertToSchema(rowType);
		System.out.println(schema.toString(true));
	}
```

This can be fixed by concating parant and child name:

```java
String fieldName = rowName + "_" + fieldNames.get(i);
```

Reply:
Okay, I will be update code to fix this problem by concating parant and child name. At the same time, i will update the test case also. Thanks!